src/System.Windows.Forms/src/System/Windows/Forms/FileDialog.cs

```diff
@@ -87,6 +87,23 @@ public bool CheckPathExists
             set => SetOption((int)Comdlg32.OFN.PATHMUSTEXIST, value);
         }
 
+        /// <summary>
+        /// <para>
+        /// Gets or sets the GUID to associate with this dialog state. Typically, state such
+        /// as the last visited folder and the position and size of the dialog is persisted
+        /// based on the name of the executable file. By specifying a GUID, an application can
+        /// have different persisted states for different versions of the dialog within the
+        /// same application (for example, an import dialog and an open dialog).
+        /// </para>
+        /// <para>
+        /// This functionality is not available if the old-style dialog is used.
```

Comment:
I think we should elaborate a little further here. Maybe something like
```suggestion
        /// This functionality is not available for old-style dialogs, that are displayed if an application is not using visual styles or <see cref="FileDialog.AutoUpgradeEnabled" /> set to <see langword="false" />.
```

@dotnet/dotnet-winforms @carlossanlop @gewarren any thoughts on the wording?
For the reference the dialog difference is depicted here: https://docs.microsoft.com/en-us/dotnet/core/compatibility/winforms#modernization-of-the-folderbrowserdialog

Reply:
@RussKie I like how specific that is. What do you think of this?

```cs
/// This functionality is not available if an application is not using visual styles or
/// <see cref="FileDialog.AutoUpgradeEnabled" /> set to <see langword="false" />.
```

Reply:
I'm not sure about the cref/langword syntax here (another one in this file is `<see cref='FileDialog'/>`), but I like the proposed revision with the addition of "if":

/// This functionality is not available if an application is not using visual
/// styles or if <see cref="FileDialog.AutoUpgradeEnabled" /> is set to
/// <see langword="false" />.

Reply:
Thank you, let's go with this version then.

Reply:
Updated.